javav2/example_code/s3/src/main/java/com/example/s3/express/S3DirectoriesActions.java

```diff
@@ -0,0 +1,551 @@
+                List<AvailabilityZone> zonesList = response.availabilityZones();
+
+                if (zonesList.isEmpty()) {
+                    logger.info("No availability zones found.");
+                    return CompletableFuture.completedFuture(null); // Return null if no zones are found
+                }
+
+                // Extract zone IDs
+                List<String> zoneIds = zonesList.stream()
+                    .map(AvailabilityZone::zoneId) // Get the zoneId (e.g., "usw2-az1")
+                    .toList();
+
+                // **Prompt user synchronously** and return CompletableFuture
+                return CompletableFuture.supplyAsync(() -> promptUserForZoneSelection(zonesList, zoneIds))
+                    .thenApply(selectedZone -> {
+                        // Return only the selected Zone ID (e.g., "usw2-az1")
+                        return selectedZone.zoneId();
+                    });
+            })
+            .whenComplete((result, exception) -> {
+                if (exception == null) {
+                    if (result != null) {
+                        logger.info("Selected Availability Zone ID: " + result);
+                    } else {
+                        logger.info("No availability zone selected.");
+                    }
+                } else {
+                    Throwable cause = exception.getCause();
+                    if (cause instanceof Ec2Exception) {
+                        throw new CompletionException("EC2 error while selecting availability zone: " + cause.getMessage(), cause);
+                    }
+                    throw new CompletionException("Failed to select availability zone: " + exception.getMessage(), exception);
+                }
+            });
+    }
+
+    /**
+     * Prompts the user to select an availability zone from the given list.
+     *
+     * @param zonesList the list of availability zones
+     * @param zoneIds the list of zone IDs
+     * @return the selected AvailabilityZone
+     */
+    private static AvailabilityZone promptUserForZoneSelection(List<AvailabilityZone> zonesList, List<String> zoneIds) {
+        Scanner scanner = new Scanner(System.in);
+        int index = -1;
+
+        while (index < 0 || index >= zoneIds.size()) {
+            logger.info("Select an availability zone:");
+            IntStream.range(0, zoneIds.size()).forEach(i ->
+                System.out.println(i + ": " + zoneIds.get(i)) // Display Zone IDs
+            );
+
+            logger.info("Enter the number corresponding to your choice: ");
+            if (scanner.hasNextInt()) {
+                index = scanner.nextInt();
+            } else {
+                scanner.next(); // Consume invalid input
+            }
+        }
+
+        AvailabilityZone selectedZone = zonesList.get(index);
+        logger.info("You selected: " + selectedZone.zoneId()); // Log Zone ID
+        return selectedZone;
+    }
+    public CompletableFuture<Void> setupVPCAsync() {
```

Comment:
Remove if we decide not to offer this route to the user.

Reply:
This is in Spec and Python app. https://github.com/awsdocs/aws-doc-sdk-examples/tree/main/scenarios/basics/s3express
As its a follow, we will keep it until spec is modified.